tests/test_geoprocessing.py

```diff
@@ -1044,8 +1044,8 @@ def test_zonal_stats_all_outside(self):
         new_feature.SetGeometry(new_geometry)
         layer.CreateFeature(new_feature)
 
-        x_pos = origin_x + (n+.99)
-        y_pos = origin_y - (n+.99)
+        x_pos = origin_x + (n-.01)

```

Comment:
This test was failing after the changes to zonal stats in this PR, and I'm not sure why it was passing in the first place. The original polygon is in purple and the new polygon is in brown. According to a comment above, this is meant to test "a polygon that barely intersects the upper left hand corner but is nodata." The original polygon actually intersects 3 valid pixels, with a sum result of `254 + 239 + 238 = 731`, not 0. I moved it up and left by one pixel to be in line with the intent of the test.
<img width="500" alt="Screenshot 2023-09-19 at 11 03 46" src="https://github.com/natcap/pygeoprocessing/assets/43770515/e50ca862-c692-413a-b154-0858cca46e3e">
.

Reply:
Sounds great, and thanks for the graphic!